mesa/examples/basic/schelling/agents.py

```diff
@@ -20,11 +19,22 @@ def step(self) -> None:
             self.pos, moore=True, radius=self.model.radius
         )
 
-        # Count similar neighbors
-        similar = sum(neighbor.type == self.type for neighbor in neighbors)
+        # Filter out empty cells
+        similar_neighbors = [
+            neighbor
+            for neighbor in neighbors
+            if hasattr(neighbor, "type") and neighbor.type == self.type
+        ]
+        total_neighbors = [
+            neighbor for neighbor in neighbors if hasattr(neighbor, "type")
```

Comment:
you iterate twice, with one check being done in both. You can make this more efficient by looping only once.